api/views.py

```diff
@@ -1,69 +1,194 @@
+    failed_at__lte = django_filters.IsoDateTimeFilter(name='failed_at', lookup_expr='lte')
+
+    class Meta:
+        model = Task
+        fields = ['id',
+                  'task_def',
+                  'status',
+                  'worker_id',
+                  'locked_at',
+                  'priority',
+                  'unique',
+                  'run_at',
+                  'started_at',
+                  'completed_at',
+                  'failed_at',
+                  'attempts',
+                  'created_at',
+                  'updated_at']
+
+class TaskList(generics.ListCreateAPIView):
+    permission_classes = (TaskServicePermission,)
+    queryset = Task.objects.all()
+    serializer_class = TaskSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filter_class = TaskFilter
+    ordering_fields = ('name',
+                       'locked_at',
+                       'run_at',
+                       'started_at',
+                       'completed_at',
+                       'failed_at',
+                       'attempts',
+                       'created_at',
+                       'updated_at')
+    ordering = ('id',)
+
+class TaskRetrieveUpdate(generics.RetrieveUpdateAPIView):
+    permission_classes = (TaskServicePermission,)
+    queryset = Task.objects.all()
+    serializer_class = TaskSerializer
+    lookup_field = 'id'
+
+class PullQueue(APIView):
+    permission_classes = (QueuePullPermission,)
+
+    def get(self, request, format=None):
+        if 'tasks' not in request.query_params:
+            raise ParseError('`tasks` query parameter required')
+
+        if 'worker_id' not in request.query_params:
+            raise ParseError('`worker_id` query parameter required')
+
+        if 'limit' in request.query_params:
+            try:
+                limit = int(request.query_params['limit'])
+            except ValueError:
+                raise ParseError('`limit` query parameter must be an integer')
+        else:
+            limit = 1
+
+        if limit < 1 or limit > 10:
+            raise ParseError('`limit` must be between 1 and 10')
+
+        raw_tasks = queue.get_tasks(request.query_params.getlist('tasks'),
+                                    request.query_params['worker_id'],
+                                    limit)
+
+        tasks = []
+        for task in raw_tasks:
+            serializer = TaskSerializer(task)
+            tasks.append(serializer.data)
+
+        return Response(tasks)
+
+class TouchTask(APIView):
+    permission_classes = (TaskServicePermission,)
+
+    def post(self, request, id):
+        if 'timeout' in request.query_params:
+            try:
+                timeout = int(request.query_params['timeout'])
+            except ValueError:
+                raise ParseError('`timeout` query parameter must be an integer')
+        else:
+            timeout = 600
+
+        if timeout < 0 or timeout > 86400:
```

Comment:
Wow. I just learned this too!

I was worried that this was taking advantage of the fact that `0 < timeout` was first evaluating to True which was `< 86400`.

I played around with it a bit though, and it seems legit:
`>>> if not -10 < -5 < 0:
...   print("asdf")
... 
>>> if not (-10 < -5) < 0:
...   print("asdf")
... 
asdf
>>> 
`

Cool!